calender.rb

```diff
@@ -0,0 +1,75 @@
+# global variables
+$days_abbrevations = %w[S M T W T F S]
+$days_start_index = {"Sunday" => 0, "Monday" => 1, "Tuesday" => 2, "Wednesday" => 3, "Thursday" => 4, "Friday" => 5, "Saturday" => 6}
+#dateObj.strftime('%A')
+
+class Calender
+    def initialize()
+        @data = Hash.new
+    end
+    
+    def add_event(date, event)
+        @data[date.year] = Hash.new if !@data[date.year]
+        @data[date.year][date.month] = Hash.new if !@data[date.year][date.month]
+        @data[date.year][date.month][date.day] = Array.new if !@data[date.year][date.month][date.day]
+        
+        @data[date.year][date.month][date.day] << event
+
+        print "\nEvent Added Succesfully\n\n"
+    end
+
+    def del_event(date, title)
+        @data[date.year][date.month][date.day].delete_if {|obj| obj.title==title}
+        print "\nEvent Deleted Successfully\n\n"
+    end
+
+    def edit_event(date, title, new_title, new_details)
+        @data[date.year][date.month][date.day].map {|obj| obj.title, obj.details = new_title, new_details if obj.title==title}
+        print "\nEvent Edited Successfully\n\n"
+    end
+
+    def print_events_details(date)
+        @data[date.year][date.month][date.day].each_with_index {|obj, index| puts "\n>> Event: #{index+1} <<\nTitle: #{obj.title}Details: #{obj.details}"}
+        print "\n"
+    end
+
+    def days_in_month(year, month)
+        Date.new(year, month, -1).day
+    end      
+
+    def print_month(date)
+        start_weekday_of_month = date.strftime('%A')
+        month_start_index = $days_start_index[start_weekday_of_month]
+
+        puts "\n\n The Day shown with color has event/s on that day\n\n"
+        puts "#{Date::MONTHNAMES[date.month]} #{date.year}".center(35)
+        puts "  #{$days_abbrevations[0]}    #{$days_abbrevations[1]}    #{$days_abbrevations[2]}    #{$days_abbrevations[3]}    #{$days_abbrevations[4]}    #{$days_abbrevations[5]}    #{$days_abbrevations[6]}  \n"
+        total_days = days_in_month(date.year, date.month) 
+
+        puts "\n"
+        (total_days+month_start_index).times do |i|
+            if i<month_start_index
```

Comment:
I guess you haven't installed Rubocop yet. Fix the spacing issues throughout the PR.

```suggestion
            if i < month_start_index
```